### PR TITLE
Add tests for coverage in envelope.go

### PR DESCRIPTION
**Pull request by adityasaky**
**Fixes issue:**

N/A

**Description:**

Adds some failure test cases to bump our coverage back up.

**Please verify and check that the pull request fulfills the following
requirements:**

- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: in_toto/envelope_test.go [hunk 1]
```diff
@@ -176,6 +176,16 @@ func TestEnvelopeVerifySignature(t *testing.T) {
...
 	})
+
+	t.Run("invalid key", func(t *testing.T) {
+		key := Key{
+			KeyID:   "invalid",
+			KeyType: "invalid",
+		}
+
+		err := env.VerifySignature(key)
+		assert.NotNil(t, err)
```
Comment (pxp928) [inline in_toto/envelope_test.go:187]: nit: hmm...just a suggestion would be to check the error you are expecting to come back and not something else.

File: in_toto/envelope_test.go [hunk 2]
```diff
@@ -192,12 +202,31 @@ func TestEnvelopeSign(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if err := env.Sign(key); err != nil {
-		t.Error(err)
-	}
+	t.Run("valid ed25519 key", func(t *testing.T) {
+		if err := env.Sign(key); err != nil {
+			t.Error(err)
```
Comment (pxp928) [inline in_toto/envelope_test.go:207]: nit: add a return if there is an error as you don't want the test to continue. or a `t.Fatal`

**Review by pxp928 — APPROVED** [review]
LGTM, couple of suggestions.

**Comment by adityasaky** [comment]
Done, thanks!

File: in_toto/envelope_test.go [hunk 3]
```diff
@@ -192,12 +202,31 @@ func TestEnvelopeSign(t *testing.T) {
...
+	t.Run("invalid key", func(t *testing.T) {
+		key := Key{
+			KeyID:   "invalid",
+			KeyType: "invalid",
+		}
 
-	assert.Equal(t, "be6371bc627318218191ce0780fd3183cce6c36da02938a477d2e4dfae1804a6", env.envelope.Signatures[0].KeyID)
-	assert.Equal(t, "HeacKZDQD+EIYz1dLJ2NpXxcG70tn62BOzcxnAArFSKJcWIL0qcyzvdtpSJQ0pOyq8lBxMk5nIRO0Kr89SZoBA==", env.envelope.Signatures[0].Sig)
+		err := env.Sign(key)
+		assert.NotNil(t, err)
```
Comment (pxp928) [inline in_toto/envelope_test.go:221]: similar here. Again I am being very nit-picky lol

File: in_toto/envelope_test.go [hunk 4]
```diff
@@ -192,12 +202,31 @@ func TestEnvelopeSign(t *testing.T) {
...
+	})
+
+	t.Run("invalid payload", func(t *testing.T) {
+		env.envelope.Payload = "abcdef"
+
+		err := env.Sign(key)
+		assert.NotNil(t, err)
```
Comment (pxp928) [inline in_toto/envelope_test.go:228]: and here

**Review by pxp928 — APPROVED** [review]

**Comment by adityasaky** [comment]
Nah, these are good!

**Pull request merged**